src/hooks/queries/useEvmFees.ts

```diff
@@ -0,0 +1,68 @@
+import type { ChainId } from '@shapeshiftoss/caip'
+import { skipToken, useQuery } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { reactQueries } from 'react-queries'
+import { selectEvmFees } from 'react-queries/selectors'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import {
+  assertGetEvmChainAdapter,
+  getFeesWithWallet,
+  type GetFeesWithWalletArgs,
+  isGetFeesWithWalletArgs,
+} from 'lib/utils/evm'
+import {
+  selectFeeAssetByChainId,
+  selectMarketDataByAssetIdUserCurrency,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type UseEvmFeesProps = Omit<
+  GetFeesWithWalletArgs,
+  'wallet' | 'adapter' | 'accountNumber' | 'data'
+> & {
+  accountNumber: number | undefined
+  data: string | undefined
+  chainId: ChainId
+  enabled?: boolean
+  staleTime?: number
+  refetchInterval?: number | false | undefined
+  refetchIntervalInBackground?: boolean
+}
+
+export const useEvmFees = (props: UseEvmFeesProps) => {
+  const wallet = useWallet().state.wallet
+
+  const {
+    enabled = true,
+    staleTime,
+    refetchInterval,
+    refetchIntervalInBackground,
+    ...input
+  } = props
+
+  const adapter = useMemo(() => assertGetEvmChainAdapter(input.chainId), [input.chainId])
+
+  const feeAsset = useAppSelector(state => selectFeeAssetByChainId(state, input.chainId))
+  const feeAssetMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, feeAsset?.assetId ?? ''),
+  )
+
+  const getFeesWithWalletInput = useMemo(
+    () => ({ ...input, adapter, wallet }),
+    [adapter, input, wallet],
+  )
+
+  const query = useQuery({
+    queryKey: reactQueries.common.evmFees(input).queryKey,
+    queryFn:
+      isGetFeesWithWalletArgs(getFeesWithWalletInput) && enabled
+        ? () => getFeesWithWallet(getFeesWithWalletInput)
+        : skipToken,
```

Comment:
They say you can't have your cake and eat it but looks like you can - abstractiong this into a query allows us to do the type narrowing on the left with `isGetFeesWithWalletArgs(getFeesWithWalletInput) && enabled`, while consumers can pass an `enabled` and now be concerned about type narrowing, and really only pass `enabled` for programmatic enabling/disabling (or none to leverage the default enabled state, assuming the input is correct with the optionals that have been type-narrowed).

This has been authored after many hours and many many iterations that ended up in correct type narrowing, but:

1. crazy boilerplate for all `evmFees()` consumers
2. most importantly, our `isXyzEnabled` consts were all doing the same thing, mixing type narrowing *and* programmatic enabling/disabling, which made things harder to grasp

This fixes both